ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+
+
+##### *proxy-send-timeout:*
+
+Sets the timeout in seconds for [transmitting a request to the proxied server](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout). The timeout is set only between two successive write operations, not for the transmission of the whole request.
+
+
+##### *resolver:*
+
+Configures name servers used to [resolve](http://nginx.org/en/docs/http/ngx_http_core_module.html#resolver) names of upstream servers into addresses
+
+
+##### *server-name-hash-max-size:*
+
+Sets the maximum size of the [server names hash tables](http://nginx.org/en/docs/http/ngx_http_core_module.html#server_names_hash_max_size) used in server names, map directive’s values, MIME types, names of request header strings, etc.
+http://nginx.org/en/docs/hash.html
+
+
+##### *server-name-hash-bucket-size:*
+
+Sets the size of the bucker for the server names hash tables
+http://nginx.org/en/docs/hash.html
+http://nginx.org/en/docs/http/ngx_http_core_module.html#server_names_hash_bucket_size
+
+##### *ssl-buffer-size:*
+
+Sets the size of the [SSL buffer](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_buffer_size) used for sending data.
+4k helps NGINX to improve TLS Time To First Byte (TTTFB)
+https://www.igvita.com/2013/12/16/optimizing-nginx-tls-time-to-first-byte/
+
+##### *ssl-ciphers:*
+
+Sets the [ciphers](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_ciphers) list to enable. The ciphers are specified in the format understood by the OpenSSL library
+The default cipher list is: `ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA`
+
+##### *ssl-protocols:*
+
+Sets the [SSL protocols](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols) to use.
+The default is: `TLSv1 TLSv1.1 TLSv1.2`
+
+TLSv1 is enabled to allow old clients like:
+- [IE 8-10 / Win 7](https://www.ssllabs.com/ssltest/viewClient.html?name=IE&version=8-10&platform=Win%207&key=113)
+- [Java 7u25](https://www.ssllabs.com/ssltest/viewClient.html?name=Java&version=7u25&key=26)
+
+If you dont need to support this clients please remove TLSv1
```

Comment:
maybe we can have a "lowest protocol" switch that will allow people to dynamically update this list by simply modifying all ingresses or the rc/pods of the ingress controller? this feels like something we can do without a complete redeployment, but it's not a huge deal